frontend/src/scenes/session-recordings/player/inspector/inspectorListFiltering.ts

```diff
@@ -0,0 +1,221 @@
+}
+
+function isPostHogEvent(item: InspectorListItem): boolean {
+    return (isEvent(item) && item.data.event.startsWith('$')) || isPostHogMobileEvent(item)
+}
+
+function isNetworkEvent(item: InspectorListItem): item is InspectorListItemPerformance {
+    return item.type === SessionRecordingPlayerTab.NETWORK
+}
+
+function isOfflineStatusChange(item: InspectorListItem): item is InspectorListOfflineStatusChange {
+    return item.type === 'offline-status'
+}
+
+function isBrowserVisibilityEvent(item: InspectorListItem): item is InspectorListBrowserVisibility {
+    return item.type === 'browser-visibility'
+}
+
+function isNavigationEvent(item: InspectorListItem): boolean {
+    return isNetworkEvent(item) && ['navigation'].includes(item.data.entry_type || '')
+}
+
+function isNetworkError(item: InspectorListItem): boolean {
+    return isNetworkEvent(item) && (item.data.response_status || -1) >= 400
+}
+
+function isSlowNetwork(item: InspectorListItem): boolean {
+    return isNetworkEvent(item) && (item.data.duration || -1) >= 1000
+}
+
+function isEvent(item: InspectorListItem): item is InspectorListItemEvent {
+    return item.type === SessionRecordingPlayerTab.EVENTS
+}
+
+function isPageviewOrScreen(item: InspectorListItem): boolean {
+    return isEvent(item) && ['$pageview', '$screen'].includes(item.data.event)
+}
+
+function isAutocapture(item: InspectorListItem): boolean {
+    return isEvent(item) && item.data.event === '$autocapture'
+}
+
+function isConsoleEvent(item: InspectorListItem): item is InspectorListItemConsole {
+    return item.type === SessionRecordingPlayerTab.CONSOLE
+}
+
+function isConsoleError(item: InspectorListItem): boolean {
+    return isConsoleEvent(item) && item.data.level === 'error'
+}
+
+function isException(item: InspectorListItem): boolean {
+    return isEvent(item) && item.data.event === '$exception'
+}
+
+function isErrorEvent(item: InspectorListItem): boolean {
+    return isEvent(item) && item.data.event.toLowerCase().includes('error')
+}
+
+function isDoctorEvent(item: InspectorListItem): item is InspectorListItemDoctor {
+    return item.type === 'doctor'
+}
+
+export function filterInspectorListItems({
+    allItems,
+    tab,
+    miniFiltersByKey,
+    showMatchingEventsFilter,
+    showOnlyMatching,
+    windowIdFilter,
+}: {
+    allItems: InspectorListItem[]
+    tab: SessionRecordingPlayerTab
+    miniFiltersByKey: {
+        [key: string]: SharedListMiniFilter
+    }
+    showMatchingEventsFilter: boolean
+    showOnlyMatching: boolean
+    windowIdFilter: string | null
+}): InspectorListItem[] {
+    const items: InspectorListItem[] = []
+
+    const shortCircuitExclude = (item: InspectorListItem): boolean =>
+        isNetworkEvent(item) && item.data.entry_type === 'paint'
+
+    const inspectorTabFilters: Record<SessionRecordingPlayerTab, (item: InspectorListItem) => boolean> = {
+        [SessionRecordingPlayerTab.ALL]: (item: InspectorListItem) => {
```

Comment:
the mini-filters aren't shared between tabs so we can first choose what tab is active
and then find the first valid match
more than one filter can be enabled but we only care if any one filter is enabled and matches